pyspedas/utilities/download.py

```diff
@@ -552,17 +708,33 @@ def download(
             else:
                 local_path_to_search = local_path
 
-            for dirpath, dirnames, filenames in os.walk(local_path_to_search):
-                local = local_file[local_file.rfind("/") + 1 :]
+            local = local_file[local_file.rfind("/") + 1 :]
+
+            # find matching files from URI
+            if is_fsspec_uri(local_path_to_search):
+                protocol, path = local_path_to_search.split("://")
+                fs = fsspec.filesystem(protocol, anon=False)
+
                 if not regex:
-                    matching_files = fnmatch.filter(filenames, local)
+                    matching_files = list(filter(lambda f: local in f, fs.find(local_path_to_search)))
                 else:
-                    reg_expression = re.compile(local)
-                    matching_files = list(filter(reg_expression.match, filenames))
+                    reg_expression = re.compile(local_path_to_search)
+                    matching_files = list(filter(reg_expression.match, fs.find(local_path_to_search)))
 
                 for file in matching_files:
-                    # out.append(os.path.join(dirpath, file))
-                    temp_out.append(os.path.join(dirpath, file))
+                    temp_out.append(protocol + "://" + file)
+                    logging.info("Streaming from local URI (download-failed): " + temp_out[-1])
```

Comment:
Can we just remove the "download-failed" part of the log? It's a bit confusing if it's just streaming from local.